server/models/meshery_controllers.go

```diff
@@ -36,146 +36,157 @@ const (
+func (mch *MesheryControllersHelper) AddMeshsynDataHandlers(ctx context.Context, k8scontext K8sContext, userID, mesheryInstanceID uuid.UUID, provider Provider) *MesheryControllersHelper {
 	// only checking those contexts whose MesheryConrollers are active
-	go func(mch *MesheryControllersHelper) {
-		mch.mu.Lock()
-		defer mch.mu.Unlock()
-		for ctxID, controllerHandlers := range mch.ctxControllerHandlersMap {
-			if _, ok := mch.ctxMeshsyncDataHandlerMap[ctxID]; !ok {
-				// brokerStatus := controllerHandlers[MesheryBroker].GetStatus()
-				// do something if broker is being deployed , maybe try again after sometime
-				brokerEndpoint, err := controllerHandlers[MesheryBroker].GetPublicEndpoint()
-				if brokerEndpoint == "" {
-					if err != nil {
-						mch.log.Warn(err)
-					}
-					mch.log.Info(fmt.Sprintf("Meshery Broker unreachable for Kubernetes context (%v)", ctxID))
-					continue
-				}
-				mch.log.Info(fmt.Sprintf("Connected to Meshery Broker (%s) for Kubernetes context (%s)", brokerEndpoint, ctxID))
-				brokerHandler, err := nats.New(nats.Options{
-					// URLS: []string{"localhost:4222"},
-					URLS:           []string{brokerEndpoint},
-					ConnectionName: MesheryServerBrokerConnection,
-					Username:       "",
-					Password:       "",
-					ReconnectWait:  2 * time.Second,
-					MaxReconnect:   60,
-				})
-				if err != nil {
-					mch.log.Warn(err)
-					mch.log.Info(fmt.Sprintf("MeshSync not configured for Kubernetes context (%v) due to '%v'", ctxID, err.Error()))
-					continue
-				}
-				mch.log.Info(fmt.Sprintf("Connected to Meshery Broker (%v) for Kubernetes context (%v)", brokerEndpoint, ctxID))
-				token, _ := ctx.Value(TokenCtxKey).(string)
-				msDataHandler := NewMeshsyncDataHandler(brokerHandler, *mch.dbHandler, mch.log, provider, userID, connectionID, mesheryInstanceID, token)
-				err = msDataHandler.Run()
-				if err != nil {
-					mch.log.Warn(err)
-					mch.log.Info(fmt.Sprintf("Unable to connect MeshSync for Kubernetes context (%s) due to: %s", ctxID, err.Error()))
-					continue
-				}
-				mch.ctxMeshsyncDataHandlerMap[ctxID] = *msDataHandler
-				mch.log.Info(fmt.Sprintf("MeshSync connected for Kubernetes context (%s)", ctxID))
+	// go func(mch *MesheryControllersHelper) {
+	
+	ctxID := k8scontext.ID
+	if mch.ctxMeshsyncDataHandler == nil {
+		controllerHandlers := mch.ctxControllerHandlers
+
+		// brokerStatus := controllerHandlers[MesheryBroker].GetStatus()
+		// do something if broker is being deployed , maybe try again after sometime
+		brokerEndpoint, err := controllerHandlers[MesheryBroker].GetPublicEndpoint()
+		if brokerEndpoint == "" {
+			if err != nil {
+				mch.log.Warn(err)
 			}
+			mch.log.Info(fmt.Sprintf("Meshery Broker unreachable for Kubernetes context (%v)", ctxID))
+			return mch
 		}
-	}(mch)
+		brokerHandler, err := nats.New(nats.Options{
+			// URLS: []string{"localhost:4222"},
```

Comment:
Is this configured as default when Meshery version = "latest" (when user has locally built Meshery)?